jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/VirtualThreadPool.java

```diff
@@ -0,0 +1,210 @@
+    {
+        if (!VirtualThreads.areSupported())
+            throw new IllegalStateException("Virtual Threads not supported");
+    }
+
+    /**
+     * @return the name of this thread pool
+     */
+    @ManagedAttribute("name of the thread pool")
+    public String getName()
+    {
+        return _name;
+    }
+
+    /**
+     * <p>Sets the name of this thread pool, used as a prefix for the thread names.</p>
+     *
+     * @param name the name of this thread pool
+     */
+    public void setName(String name)
+    {
+        if (isRunning())
+            throw new IllegalStateException(getState());
+        if (StringUtil.isBlank(name) && name != null)
+            throw new IllegalArgumentException("Blank name");
+        _name = name;
+    }
+
+    public void setTracking(boolean tracking)
+    {
+        if (isRunning())
+            throw new IllegalStateException(getState());
+        _tracking = tracking;
+    }
+
+    public boolean isTracking()
+    {
+        return _tracking;
+    }
+
+    @ManagedAttribute("reports additional details in the dump")
+    public boolean isDetailedDump()
+    {
+        return _detailedDump;
+    }
+
+    public void setDetailedDump(boolean detailedDump)
+    {
+        _detailedDump = detailedDump;
+        if (_virtualExecutor instanceof TrackingExecutor trackingExecutor)
+            trackingExecutor.setDetailedDump(detailedDump);
+    }
+    
+    @Override
+    protected void doStart() throws Exception
+    {
+        _main = new Thread("virtual main")
+        {
+            @Override
+            public void run()
+            {
+                try (AutoLock.WithCondition l = _joinLock.lock())
+                {
+                    while (isRunning())
+                    {
+                        l.await();
+                    }
+                }
+                catch (InterruptedException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        _main.start();
+
+        _virtualExecutor = Objects.requireNonNull(StringUtil.isBlank(_name)
+            ? VirtualThreads.getDefaultVirtualThreadsExecutor()
+            : VirtualThreads.getNamedVirtualThreadsExecutor(_name));
+        if (_tracking)
+            _virtualExecutor = new TrackingExecutor(_virtualExecutor, _detailedDump);
+        addBean(_virtualExecutor);
+        super.doStart();
+    }
+
+    @Override
+    protected void doStop() throws Exception
+    {
+        super.doStop();
+        _virtualExecutor = null;
+        _main = null;
+
+        try (AutoLock.WithCondition l = _joinLock.lock())
+        {
+            l.signalAll();
+        }
+    }
+
+    @Override
+    public Executor getVirtualThreadsExecutor()
+    {
+        return _virtualExecutor;
+    }
+
+    @Override
+    public void setVirtualThreadsExecutor(Executor executor)
+    {
+        throw new UnsupportedOperationException("cannot set VirtualThreadExecutor");
```

Comment:
Why not?
Users may pass their own if so wish (e.g. shared with other components not related to Jetty), and we can wrap it.
Perhaps just have a guard for `isStarted()`?